### ISSUE TITLE
Bluetooth mouse's connection drops quickly

### ISSUE THREAD

**Issue by khenty** [linux-surface/linux-surface#1178]
[Description of the bug or feature]
I've set up my surface pro 9 to dual boot win11 and Linux Mint 21.1 running the Mate desktop. I've also installed the latest linux-surface kernel (as of 7/3/2023) and verified I have the latest bluetooth drivers. The keyboard works well, but although I can get the bluetooth mouse to connect, the connection is dropped almost immediately during normal use. For instance, I connect the mouse to bluetooth (Blueman interface), open Firefox, Firefox captures focus so that I can enter my password, and the mouse is gone. The bluetooth interface says I still have a connection, but I don't. I've used the same mouse on a Lenovo 11e laptop with the same version of Linux Mint, without a problem. I've also tried Bluetuith and different bluetooth mice, same result. The ONLY way I've been able to force the bluetooth mouse to stay connected is by also connecting a mechanical mouse via USB, pretty much defeating the reason for having a buetooth mouse. Can anyone recommend a fix for this? Thanks!

### Environment

- Hardware model: 
- Kernel version: 
- Distribution: 

<details><summary>`dmesg` output</summary>

```
please provide a copy of `dmesg` here if possible

```

</details>

**Comment by khenty** [comment]
Hardware model: surface pro 9, 
Kernel version: linux-image-6.3.7-surface
Distribution: Linux Mint 21.1 
dmesg output 
thebes:~> dmesg | grep -i bluetooth
[    7.245396] Bluetooth: Core ver 2.22
[    7.245413] NET: Registered PF_BLUETOOTH protocol family
[    7.245414] Bluetooth: HCI device and connection manager initialized
[    7.245418] Bluetooth: HCI socket layer initialized
[    7.245419] Bluetooth: L2CAP socket layer initialized
[    7.245422] Bluetooth: SCO socket layer initialized
[    7.328258] Bluetooth: hci0: Device revision is 0
[    7.328263] Bluetooth: hci0: Secure boot is enabled
[    7.328264] Bluetooth: hci0: OTP lock is enabled
[    7.328265] Bluetooth: hci0: API lock is enabled
[    7.328266] Bluetooth: hci0: Debug lock is disabled
[    7.328267] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[    7.328269] Bluetooth: hci0: Bootloader timestamp 2019.40 buildtype 1 build 38
[    7.329809] Bluetooth: hci0: Found device firmware: intel/ibt-0040-0041.sfi
[    7.329832] Bluetooth: hci0: Boot Address: 0x100800
[    7.329834] Bluetooth: hci0: Firmware Version: 107-51.22
[    7.742453] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[    7.742458] Bluetooth: BNEP filters: protocol multicast
[    7.742462] Bluetooth: BNEP socket layer initialized
[    8.808682] Bluetooth: hci0: Waiting for firmware download to complete
[    8.810222] Bluetooth: hci0: Firmware loaded in 1445713 usecs
[    8.810268] Bluetooth: hci0: Waiting for device to boot
[    8.826223] Bluetooth: hci0: Device booted in 15608 usecs
[    8.826224] Bluetooth: hci0: Malformed MSFT vendor event: 0x02
[    8.827300] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0040-0041.ddc
[    8.830233] Bluetooth: hci0: Applying Intel DDC parameters completed
[    8.833256] Bluetooth: hci0: Firmware timestamp 2022.51 buildtype 1 build 56683
[    8.901498] Bluetooth: MGMT ver 1.22
[    8.918726] Bluetooth: RFCOMM TTY layer initialized
[    8.918733] Bluetooth: RFCOMM socket layer initialized
[    8.918735] Bluetooth: RFCOMM ver 1.11
[    8.999530] Bluetooth: hci0: Bad flag given (0x1) vs supported (0x0)
[   30.486199] hid-generic 0005:000E:3412.0008: input,hidraw6: BLUETOOTH HID v4.00 Mouse [BT5.0 Mouse] on 58:ce:2a:ce:f2:b5

**Comment by loligans** [comment]
I'm having the same issue with my "New Microsoft Ergonomic" Mouse

**Comment by loligans** [comment]
I was able to determine which device was the mouse by reading the `RSSI` value. The closer to `0` means the device is close to the laptop. So I chose the MAC address closest to `0` (in my case it was between `-35` and `-45`) and it connected to the mouse. However while I am connected to the mouse it doesn't appear to be working. I'm troubleshooting it now

**Comment by yayinakisi** [comment]
Hi there,

I was having the same problem and have the fix that I hope will work for you too.
As far as I understand, what moues does is kind of auto suspending and most probably for power saving reasons.

The solution is to create udev rule.
Go to terminal and command lsusb -vt to get the usb device id. In my case the output is:
/:  Bus 04.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/4p, 10000M
    ID 1d6b:0003 Linux Foundation 3.0 root hub
/:  Bus 03.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/12p, 480M
    ID 1d6b:0002 Linux Foundation 2.0 root hub
    |__ Port 10: Dev 2, If 0, Class=Wireless, Driver=btusb, 12M
        ID 8087:0033 Intel Corp. 
    |__ Port 10: Dev 2, If 1, Class=Wireless, Driver=btusb, 12M
        ID 8087:0033 Intel Corp. 
/:  Bus 02.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/4p, 20000M/x2
    ID 1d6b:0003 Linux Foundation 3.0 root hub
/:  Bus 01.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/1p, 480M
    ID 1d6b:0002 Linux Foundation 2.0 root hub

And then create /etc/udev/rules.d/50-usb_power_saving.rules which is contains

**Comment by khenty** [comment]
Thanks! I'll give that a try and let you know. Since my posting I gave up and have been using a 2.4 Ghz USB adapter. And excuse me if you get this message twice. My reply to your email doesn't seem to have gotten through. 

Glenn

**Comment by khenty** [comment]
Seems to work on one bluetooth mouse (Tecknet), but not another (Logitech). At least I have one working bluetooth mouse now. Thanks again!

**Comment by yayinakisi** [comment]
Have you written the correct idVendor and idProduct which are device specific?

**Comment by khenty** [comment]
Yes. I think the problem lies with the Logitech mouse, as I can't get it to work on Windows 11 either, unless I install a proprietary driver.

**Comment by loligans** [comment]
I switched over to use PopOS with linux-surface kernel and my device now shows up in the list of devices. Idk how much different Ubuntu 20.04 could be from PopOS

**Comment by Mickael-Pays** [comment]
Hi,
I have a similar issue. I can't see my Microsoft Mobile Modern mouse in bluetooth list.

Hardware SP9 intel
Linux 6.5.7-surface-2 #2 SMP PREEMPT_DYNAMIC Sun Oct 29 12:36:11 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
Ubuntu 22.04.3 LTS

Someone can help me ?
